src/Connector.js

```diff
@@ -221,18 +221,16 @@ async function removeDeviceFromOrion(orionUrl, id) {
 
   const promises = sensors.map(async (sensor) => {
     url = `${orionUrl}/v2/entities/${sensor.id}`;
-    await request.delete({ url, headers, json: true });
+    return request.delete({ url, headers, json: true });
   });
 
   await Promise.all(promises);
 }
 
 async function subscribeToEntities(client, devices) {
   const promises = devices.map(async (device) => {
-    await client.subscribe(`/default/${device.id}/cmd`);
-    device.schema.map(async (sensor) => {
-      await client.subscribe(`/${device.id}/${sensor.sensorId}/cmd`);
-    });
+    client.subscribe(`/default/${device.id}/cmd`);
```

Comment:
```
await Promise.all(device.schema.map(async sensor => client.subscribe(`/${device.id}/${sensor.sensorId}/cmd`)));
return client.subscribe(`/default/${device.id}/cmd`);
```

Reply:
You should have two Promise.All

Is it have some precedence in subscriptions?

Reply:
Do you mean, nested `Promise.all()`? :eyes:

Reply:
I think there is no precedence between them, so call the `Promise.all()` twice may not be applied to this context.

Reply:
Despite having no precedence, the solution suggested by @ramonhpr (already pushed, also shown below) makes sense: both Device and Schema operations are asynchronous iterable operations. To avoid returning **undefined**, considering there is no precedence, we then `return  the device subscription (which will only be done once per device). 

It wouldn't make sense having two `Promise.all()` in cases such as `listDevices()`, which the iterable operation (within `Promise.all()`) isn't asynchronous -- which doesn't return a promise. 

```suggestion
  await Promise.all(devices.map(async ({ id, schema }) => {
    Promise.all(schema.map(async ({ sensorId }) => client.subscribe(`/${id}/${sensorId}/cmd`)));
    return client.subscribe(`/default/${id}/cmd`);
  }));
```

Reply:
As discussed earlier, w/ @ramonhpr, it seems that a single, outer, `Promise.all()` is enough to ensure we run the asynchronous operations in `subscribeToEntities()` concurrently :smile:

Reply:
Yes, it is. 😃

Reply:
@ramonhpr, done :smile: